### ISSUE TITLE
Why is the optimizer defined twice in the tuner-subclassing tutorial?

### ISSUE THREAD

**Issue by bw0248** [keras-team/keras-tuner#245]
I'm reading the tuner-subclassing tutorial for custom training loops on keras-team.github.io ([link](https://keras-team.github.io/keras-tuner/tutorials/subclass-tuner/))
In the end-to-end example I don't understand why the optimizer is defined twice. 
Once inside the `build_model` function:

    optimizer = hp.Choice('optimizer', ['adam', 'sgd'])
    model.compile(optimizer, loss='sparse_categorical_crossentropy', metrics=['accuracy'])
    return model

This make sense to me, but then inside `trial_run` an optimizer is defined again:

    lr = hp.Float('learning_rate', 1e-4, 1e-2, sampling='log', default=1e-3)
    optimizer = tf.keras.optimizers.Adam(lr)

This second optimizer is then used during `run_train_step`

    with tf.GradientTape() as tape:
        # calculate loss
        ...
    optimizer.apply_gradients(zip(gradients, model.trainable_variables))

Wouldn't this mean that the defined Hyperparameter for the optimizer inside `build_model` is never used? The way I understand the example, the optimizer used for `apply_gradients` inside `run_train_step` wil always be `Adam` with varying learning rates, but never `SGD`.
Am I thinking wrong here?

**Comment by wen036** [comment]
I found this confusing as well. My guess is that might be an error.
